### PR TITLE
Add Nix Flake to support building from source for NixOS

### PR DESCRIPTION
**Pull request by jue89**
Tested with the following command:

```
nix --experimental-features "nix-command flakes" run github:jue89/dune3d
```

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        pkgs.wrapGAppsHook
+        pythonEnv
+      ];
+      buildInputs = [
+        pkgs.cmake
+        pkgs.eigen
+        pkgs.glm
+        pkgs.gtkmm4
+        pkgs.libepoxy
+        pkgs.librsvg
```
Comment (carrotIndustries) [inline flake.nix:34]: rsvg is only a build-time dependency

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      nativeBuildInputs = [
+        pkgs.meson
+        pkgs.ninja
+        pkgs.gobject-introspection
+        pkgs.pkg-config
+        pkgs.wrapGAppsHook
+        pythonEnv
+      ];
+      buildInputs = [
+        pkgs.cmake
```
Comment (carrotIndustries) [inline flake.nix:29]: cmake is only required at build time

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        pkgs.eigen
+        pkgs.glm
```
Comment (carrotIndustries) [inline flake.nix:31]: build-time only since these are header-only

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        pkgs.gtkmm4
+        pkgs.libepoxy
+        pkgs.librsvg
+        pkgs.libuuid
+        pkgs.mimalloc
+        pkgs.opencascade-occt
+        pkgs.range-v3
```
Comment (carrotIndustries) [inline flake.nix:38]: build-time only since it's header-only

File: flake.nix (hunk 3 above)
Comment (jue89) [reply flake.nix:32]: If I get the nixpkgs builder right, it's still considered to be part of `buildInputs`. I haven't found a single package that's using `glm` in the `nativeBuildInputs`

File: flake.nix (hunk 4 above)
Comment (jue89) [reply flake.nix:38]: Dito. It's part if the build and not a helper to build Dune3D.

**Comment by carrotIndustries** [comment]
Ok, I'll merge it once the fixup commit is squashed.

**Comment by jue89** [comment]
There you go :)

**Comment by jue89** [comment]
Cool! Thank you for your super fast review and merge :)

**Pull request merged**